source/validate_instruction.cpp

```diff
@@ -131,6 +131,68 @@ spv_result_t CapCheck(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Checks that the number of characters in a string doesn't exceed the limit.
+spv_result_t LimitCheckString(ValidationState_t& _,
+                              const spv_parsed_instruction_t* inst) {
+  // By definition, num_words of an operand is a 16-bit unsigned which will not
+  // exceed the limit of characters for a string literal.
+  // This function checks that a NULL termination must be found before reaching
+  // the end of the string words.
+  for (int i = 0; i < inst->num_operands; ++i) {
+    auto operand = inst->operands[i];
+    if (SPV_OPERAND_TYPE_LITERAL_STRING == operand.type) {
+      bool null_terminated = false;
```

Comment:
It seems that the compiler does not like `strlen_s`

`no member named 'strnlen_s' in namespace 'std'`

adding `#include <string.h>` and `#include <string>` did not help.
I also couldn't find the function in cplusplus.com (although it is in cppreference.com). maybe it's only supported my MS?